lib/copy_color_spinor_offset.cu

```diff
@@ -0,0 +1,72 @@
+#include <color_spinor_field.h>
+#include <color_spinor_field_order.h>
+
+#include <kernels/copy_field_offset.cuh>
+
+#include <instantiate.h>
+
+namespace quda
+{
+
+  template <class Field, class Element, class F>
+  void copy_color_spinor_offset(Field &out, const Field &in, CommKey offset, QudaPCType pc_type)
+  {
+    F out_accessor(out);
+    F in_accessor(in);
+    if (pc_type == QUDA_4D_PC) {
+      using Arg = CopyFieldOffsetArg<Field, Element, F, QUDA_4D_PC>;
+      Arg arg(out_accessor, out, in_accessor, in, offset);
+      CopyFieldOffset<Arg> copier(arg, in);
+    } else {
```

Comment:
Same comment as above about querying other `pc_type`s instead of just using `else`, adding a `static_assert`.

Reply:
Ah, not a `static_assert` here, but an `errorQuda` to catch the true `else`